### DISCUSSION TITLE
With Fo-Dicom, how do you make a case-insensitive CFindRequest?

### DISCUSSION THREAD

**Discussion by robertkjr3d** [fo-dicom/fo-dicom#1342] [Q&A] [answered]
I have asked this same question at stackoverflow, but I'm losing confidence that it will be answered there.
I can make these great worklist queries like this one...

```
//Worklist MWL PATIENTID query
var cf = DicomCFindRequest.CreateWorklistQuery();
cf.Dataset.AddOrUpdate(DicomTag.PatientID, szPatientIDsearch);
cf.OnResponseReceived = (DicomCFindRequest rq, DicomCFindResponse rp) =>
{
       if (rp.HasDataset)
       {
           worklistItems.Add(rp.Dataset);
       }
};
Dicom.Network.Client.DicomClient client = new Dicom.Network.Client.DicomClient(IPAddress, mwlserver.port, false, Preferences.SendingAETitle, mwlserver.AETitle, 5000, 10000, 50, 5);
await client.AddRequestAsync(cf);
await client.SendAsync();
```
But how do you make the PatientID, or a Patientname case insensitive? Is there a DicomTag to add?

**Comment by gofal** [comment, marked as answer]
No, there is no DICOM Tag you can add. DICOM per se is case sensitive. If you send a query to the DICOM server, then it is up the implementation on the server side, of they treat the query in a case sensitive or insensitive way. You cannot change that with DICOM methods. You have to ask the vendor of the PACS, if they can do some configuration to make the query case insensitive. or maybe they give you some hint what you have to do to make the query case insensitive.

**Reply by gofal** [reply]
some reference: https://groups.google.com/g/comp.protocols.dicom/c/KdHGOhPmYj4
